tests/test_setuptools_pep517.py

```diff
@@ -15,14 +15,15 @@
 def test_pep517_sdist():
     correct_metadata = textwrap.dedent(
         """\
-        Metadata-Version: 2.1
```

Comment:
I think you will need to conditionalize this on the version of `setuptools` otherwise the Fedora tests would fail.

Reply:
It's only checking fields it knows about, so removing it just makes it not matter. (This is what the setuptools tests do in the linked PR)

Reply:
(It's checked by `assert metadata_set <= pkg_info_contents`)

Reply:
Oof looks very round-about way to compare them and flaky. But it seems that indeed it does what it says it would so :shrug: